### PR TITLE
Replace asset check with amount check for asset transfer deep links

### PR DESCRIPTION
**Pull request by mitsinsar**

### CODE REVIEW

File: versions.gradle [hunk 1]
```diff
@@ -60,6 +60,6 @@ ext {
 
     peraApplication = [
             versionCode: 118,
-            versionName: "5.10.13"
+            versionName: "90.20.772-1"
```
Comment (ehanoc) [inline versions.gradle]: What does this number represent?

File: app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt [hunk 2]
```diff
@@ -196,7 +196,7 @@ sealed class BaseDeepLink {
 
             override fun doesDeeplinkMeetTheRequirements(rawDeepLink: RawDeepLink): Boolean {
                 return with(rawDeepLink) {
-                    val doesDeeplinkHaveAssetTransferQueries = accountAddress != null && assetId != null
+                    val doesDeeplinkHaveAssetTransferQueries = accountAddress != null && amount != null
```
Comment (ehanoc) [inline app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt]: Not need for asset check? We assume it's always algo?

File: versions.gradle (hunk 1 above)
Comment (mitsinsar) [reply versions.gradle]: 90 means internal release
20 is sprint
772 is ticket number
- 1 is the iteration of the ticket.

File: app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt (hunk 2 above)
Comment (mitsinsar) [reply app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt]: Checking amount is enough to see that this is an asset transfer deep link. And while creating the asset transfer deep link object, we get asset id `if exists`,  if not, we say this is an Algo transfer deep link

File: versions.gradle (hunk 1 above)
Comment (ehanoc) [reply versions.gradle]: Okay, why is it being updated here? versioning should only be updated for releases and as part of the building step ideally. no?

File: app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt (hunk 2 above)
Comment (ehanoc) [reply app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt]: > while creating the asset transfer deep link object

While creating, not verifying? Sorry, not sure i understood how we check if it's a simple algo transfer or an asset transfer here.

File: versions.gradle (hunk 1 above)
Comment (mitsinsar) [reply versions.gradle]: We have two CI/CD trigger patterns;
`*.*.*-*` -> Triggers internal release workflow
`*.*.*` -> Triggers store release workflow

This fix will be a hot fix, that's why I triggered internal test release immediately.

File: app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt (hunk 2 above)
Comment (mitsinsar) [reply app/src/main/java/com/algorand/android/modules/deeplink/domain/model/BaseDeepLink.kt]: In this step, we don't need to check if this is an asset transfer or Algo transfer. We don't have two different asset transfer models like `AssetTransferDeepLink` and `AlgoTransferDeepLink`, we just have `AssetTransferDeepLink`. That's why it is enough to check if this is a `transfer` (has account address and amount) deep link or not. 

And while creating `AssetTransferDeepLink` object, we simply check if deep link has `asset` query or not. If there is, we parse it, if not, we say this is an Algo transfer and pass Algo asset id.

Something like;
```
AssetTransferDeepLink(
    assetId = if (rawDeeplink.assetId) != null rawDeeplink.assetId else ALGO_ASSET_ID,
    ...
)
```

**Review by yasin-ce — APPROVED** [review]

**Pull request merged**